compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/PruneNonEscapingScopes.ts

```diff
@@ -7,8 +7,8 @@
 
 import {CompilerError} from '../CompilerError';
 import {
+  DeclarationId,
```

Comment:
todo: leave a comment on why the change here. This pass doesn't model control flow, only data flow, so it relies on LeaveSSA having assigning a single IdentifierId to all SSA instances - this means that regardless of control flow, any data that flows into a given program variable is captured. This is conservative but correct - we could build a more precise version using SSA ids and following control flow, but using DeclarationId preserves the previous behavior.